recipes/qtxlsxwriter/all/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(qtxlsxwriter)
+
+find_package(qt REQUIRED CONFIG)
+
+set(QTXLSXWRITER_TARGET_NAME qtxlsxwriter)
+
+include(conan_qt_executables_variables)
+set(CMAKE_AUTOMOC ON)
+
+file(GLOB QTXLSXWRITER_HEADERS LIST_DIRECTORIES=false source_subfolder/src/xlsx/*.h)
+file(GLOB QTXLSXWRITER_SOURCES LIST_DIRECTORIES=false source_subfolder/src/xlsx/*.cpp)
+
+add_library(${QTXLSXWRITER_TARGET_NAME}
+	${QTXLSXWRITER_HEADERS}
+	${QTXLSXWRITER_SOURCES}
+)
+
+target_include_directories(${QTXLSXWRITER_TARGET_NAME}
+	PRIVATE
+		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source_subfolder/src/xlsx>
+		# qtxlsxwriter uses qt private headers
+		$<BUILD_INTERFACE:${QT_ROOT}/include/QtCore/5.15.2>
+		$<BUILD_INTERFACE:${QT_ROOT}/include/QtGui/5.15.2>
```

Comment:
I think we can use the variables provided by the qt package. They look quite reliable because they are needed for [`automoc`](https://cmake.org/cmake/help/v3.20/prop_tgt/AUTOMOC.html#qt-version-detection), for example. [9f23cf9](https://github.com/conan-io/conan-center-index/pull/4838/commits/9f23cf9ccff6a2e7d2a77d770432d7096a2542b8)